lib/overcommit/utils.rb

```diff
@@ -124,15 +124,14 @@ def execute(args)
       # which we do not need to know the result.
       #
       # @param args [Array<String>]
-      # @return [Thread] thread watching the resulting child process
+      # @return [ChildProcess] thread watching the resulting child process
       def execute_in_background(args)
         if args.include?('|')
           raise Overcommit::Exceptions::InvalidCommandArgs,
                 'Cannot pipe commands with the `execute_in_background` helper'
         end
 
-        # Dissociate process from parent's input/output streams
-        Process.detach(Process.spawn({}, *args, [:in, :out, :err] => '/dev/null'))
+        Subprocess.spawn_in_background(args)
```

Comment:
We can detach the `ChildProcess` using `process.detach = true` before executing `process.start`. I only omitted that line because `process.exited?` [throws an error](https://github.com/jarib/childprocess/issues/84) when `process` is detached, which caused the spec to fail. Didn't consider the problem of zombie processes.